src/shared/lib/injectors.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { ComponentType, useMemo, PropsWithChildren } from 'react'
+import { SerializedUser } from '../../cloud/interfaces/db/user'
+import { usePage } from '../../cloud/lib/stores/pageStore'
+
+export function withLiveUser<
```

Comment:
We can make a hook `useLiveUser` which using the same hooks of `withLiveUser`
So I would like to use the below code.

```
export function useLiveUser() {
      const { permissions } = usePage()

    const normalizedUser = useMemo(() => {
      if (permissions == null) {
        return props.user
      }

      const permission = permissions.find((permission) => permission.user.id)
      return permission != null ? permission.user : props.user
    }, [props.user, permissions])
}

const LiveUserIcon = (props: Exclude<UserIconProps, 'user'>) => {
  const liveUser = useLiveUser()

   return <UserIcon {...props} user={liveUser}/>
}
```

By doing this, we don't have to introduce another pattern HOC.
And the hook pattern is much easier to compose with other hooks while composing HOCs are quite difficult to understand and control.)